core/templates/domain/exploration/exploration-id-validation.service.spec.ts

```diff
@@ -66,62 +64,84 @@ describe('Exploration id validation service', function() {
+      let failHandler = jasmine.createSpy('fail');
+      let explorationIds = ['0'];
+      let requestUrl = '/explorationsummarieshandler/data?' +
+        'stringified_exp_ids=' + encodeURI(JSON.stringify(explorationIds)) +
+        '&' + 'include_private_explorations=false';
+
+      explorationIdValidationService.isExpPublished('0')
+        .then(successHandler, failHandler);
+      let req = httpTestingController
+        .expectOne(requestUrl);
+      expect(req.request.method).toEqual('GET');
+      req.flush({summaries: []});
+      flushMicrotasks();
+
+      expect(successHandler).toHaveBeenCalledWith(false);
+      expect(failHandler).not.toHaveBeenCalled();
 
       // The service should respond false when the summaries array
       // contains null.
-      $httpBackend.expectGET(/.*?explorationsummarieshandler?.*/g).respond({
-        summaries: [null]
-      });
-      ExplorationIdValidationService.isExpPublished('0').then(
-        function(response) {
-          expect(response).toEqual(false);
-        });
-      $httpBackend.flush();
+      explorationIdValidationService.isExpPublished('0')
+        .then(successHandler, failHandler);
+      req = httpTestingController
+        .expectOne(requestUrl);
+      expect(req.request.method).toEqual('GET');
+      req.flush({summaries: [null]});
+      flushMicrotasks();
+
+      expect(successHandler).toHaveBeenCalledWith(false);
+      expect(failHandler).not.toHaveBeenCalled();
 
       // The service should respond false when the summaries array
       // contains more than one element.
-      $httpBackend.expectGET(/.*?explorationsummarieshandler?.*/g).respond({
-        summaries: [
-          'exp_1',
-          'exp_2'
-        ]
-      });
-      ExplorationIdValidationService.isExpPublished('0').then(
-        function(response) {
-          expect(response).toEqual(false);
-        });
-      $httpBackend.flush();
+      explorationIdValidationService.isExpPublished('0')
+        .then(successHandler, failHandler);
+      req = httpTestingController
+        .expectOne(requestUrl);
+      expect(req.request.method).toEqual('GET');
+      req.flush({summaries: [
+        'exp1',
+        'exp2'
+      ]});
+      flushMicrotasks();
+
+      expect(successHandler).toHaveBeenCalledWith(false);
+      expect(failHandler).not.toHaveBeenCalled();
     }
-  );
+    ));
 
   it('should correctly validate the valid exploration id',
-    function() {
-      $httpBackend.expectGET(/.*?explorationsummarieshandler?.*/g).respond(
-        validExpResults);
-      ExplorationIdValidationService.isExpPublished('0').then(
-        function(response) {
-          expect(response).toEqual(true);
-        });
-      $httpBackend.flush();
+    fakeAsync(()=> {
+      let successHandler = jasmine.createSpy('success');
+      let failHandler = jasmine.createSpy('fail');
+      let explorationIds = ['0'];
+      let requestUrl = '/explorationsummarieshandler/data?' +
```

Comment:
I think these can be changed to const.

Optional: These initializations seem to common in both the tests. You can move this to a beforeEach block so that the code is more "DRY".

Reply:
done!